lib/utils/isDescriptorDeclaration.cjs

```diff
@@ -14,8 +14,15 @@ const atKeywords = require('../reference/atKeywords.cjs');
 function isDescriptorDeclaration(decl) {
 	const { parent } = decl;
 
-	if (parent && typeGuards.isAtRule(parent) && !atKeywords.nestingSupportedAtKeywords.has(parent.name.toLowerCase()))
-		return true;
+	/** @type {import('postcss').Node | undefined} */
+	let node = parent;
+
+	while (node && node.type !== 'root') {
```

Comment:
off-topic
remark

We have 4 rules with this condition; we could maybe add an `isNested` util.

Reply:
That idea could be likely. I have a rough idea for implementing more generic utilities. Like so:

```js
/** @import { Node } from 'postcss' */

/**
 * @param {Node} startNode
 * @param {(node: Node) => void | false} callback
 * @returns {void}
 */
function eachNodeUpToRoot(startNode, callback) {
	let currentNode = startNode.parent;

	while (currentNode && !isRoot(currentNode)) {
		if (callback(currentNode) === false) break;

		currentNode = currentNode.parent;
	}
}

/**
 * @param {Node} startNode
 * @param {(node: Node) => boolean} condition
 * @returns {Node | undefined}
 */
function findNodeUpToRoot(startNode, condition) {
	let found;

	eachNodeUpToRoot(startNode, (node) => {
		if (condition(node)) {
			found = node;

			return false;
		}
	});

	return found;
}
```

But I'm still unsure if adding them is worth it. Anyway, it's out of the scope of this PR. 😅

Reply:
It's a good suggestion. I've opened an issue to continue the discussion https://github.com/stylelint/stylelint/issues/8221